vulnerabilities/pagination_mixin.py

```diff
@@ -0,0 +1,58 @@
+class PaginatedListViewMixin:
+    paginate_by = 20
+    max_page_size = 100
+
+    PAGE_SIZE_CHOICES = [
+        {"value": 20, "label": "20 per page"},
+        {"value": 50, "label": "50 per page"},
+        {"value": 100, "label": "100 per page"},
+    ]
+
+    def get_paginate_by(self, queryset=None):
```

Comment:
This function handles page size from URL parameters. Sets default to 20 items, max of 100. Validates to ensure positive integers only. Returns default if invalid values provided.